scheme/deltas/071-companies-name-constraint.sql

```diff
@@ -0,0 +1,2 @@
+DROP INDEX `nameBrand` ON `Companies`; 
+CREATE UNIQUE INDEX `nameBrandType` ON `Companies`(`name`,`brandId`,`type`);
```

Comment:
Even if this change may make sense, I would not change the current unique key as it comes with some drawbacks: Currently, the company name is the only way to distinct them in emulation selector. I cannot think of any other issue right now.

Because of that, this would require some frontend work as well in order to be accepted.

Reply:
If I understand correctly what is the emulation selector I don't think any other change is needed.

After this change in the emulation selector, if I understand correctly it is the dropdown menu to select the company you want to emulate, if there are a vpbx and a retail with same name they both appear correctly in the dropdown. A vpbx icon prepends the name of the vpbx and a retail icon prepends the name of the retail.

When you select the vpbx the system correctly emulates the vpbx. When you select the retail the system correctly emulates the retail account.

Not sure if this is what you mean.

Reply:
My mistake. I'm sorry, I should have taken some time to check this out beforehand. 

It turns out that emulation selector is the only select combo with icons. The issue would be applicable to the rest of the company selectors: Outgoing routing and invoices for instance.

Reply:
I see now those other places. Thanks.

In my opinion it is better to disallow same name on retail and vpbx then. So I will remove the delta and change the captured constraint in Companies.php.

What's your opinion? Preserve the (name,brandId) constraint or change the code of those other company selectors in outgoing routing, invoices, ...?

Reply:
Adding the capability to show an icon into select options could take some time, so I would preserve the constraint too. 

Although showing country flags and stuff like that has been on our wishlist for some time now, we will be able to find some time to implement it eventually. We could take back this topic then.

Thanks for your contribution by the way :)